src/librustc_typeck/diagnostics.rs

```diff
@@ -3284,6 +3284,82 @@ impl Baz for Bar { } // Note: This is OK
 ```
 "##,
 
+E0374: r##"
+The trait `CoerceUnsized` may only be implemented for a coercion between
+structures with one field being coerced, none were found. The structure you
+are trying to implement `CoerceUnsized` for doesn't have any Unsized fields
+so it doesn't need CoerceUnsized.
+
+Example:
+
+```compile_fail
+#![feature(coerce_unsized)]
+struct Foo<T> {
+    a: i32,
+}
+
+// error: Struct `Foo` has no unsized fields that need `CoerceUnsized`.
+impl<T, U> CoerceUnsized<Foo<U>> for Foo<T> {}
+```
+
+`CoerceUnsized` is used to coerce one unsized structure into another. If the
+structure doesn't have any unsized fields, it's not an unsized structure
+and can use regular type coercion instead.
+"##,
+
+E0375: r##"
+The trait `CoerceUnsized` may only be implemented for a coercion between
+structures with one field being coerced, but multiple fields need coercions.
+The structure you are trying to implement `CoerceUnsized` for has more than
+one unsized field that would need to be coerced.
+
+Example:
```

Comment:
Same comment as above: take a look at https://github.com/rust-lang/rfcs/pull/1567 about "Example of erroneous code".